snapcraft/hooks/connect-plug-graphics-core22

```diff
@@ -0,0 +1,32 @@
+#!/bin/sh
+set -eu
+
+# Re-exec outside of apparmor confinement
+if [ -d /sys/kernel/security/apparmor ]; then
+  label="$(cat /proc/self/attr/current 2>/dev/null)"
+  if [ "$label" != "unconfined" ] && [ -n "${label##*(unconfined)}" ]; then
+    exec aa-exec -p unconfined -- "$0" "$@"
+  fi
+fi
+
+if [ -e "/etc/.lxd_generated" ]; then
+    nvidia_container_cli="${SNAP}/bin/nvidia-container-cli"
+
+    # Create a backup if it doesn't exist
+    if [ ! -f "${nvidia_container_cli}.bak" ]; then
+        cp "${nvidia_container_cli}" "${nvidia_container_cli}.bak"
+    fi
+
+    # Remove the root path frrm being `/var/lib/snapd/hostfs/` (host system)
+    # This will then allow `nvidia-container-cli` to discover userspace libs
+    # living within the snap mount namespace.
+    cat > "${nvidia_container_cli}" << EOF
```

Comment:
Rather than dynamically re-writing the file from an embedded script in a hook (which is hard to discover and reason about), I would prefer that both versions of the script exist inside the snap and we dynamically change the symlink to the active one in the hook.